AAE-37399 Add root process instance id to task

mcoune opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 The commit message follows our guidelines
     - 👷‍♂️ Tests for the changes have been added (for bug fixes / features)
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.
-->

<!--
(check one with "x") one of the following
-->

**What kind of change does this PR introduce?**

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

## Description

Extend task model with root process instance id to use it in analytics and openSearch

- Issue Link:
[JIRA TICKET](https://hyland.atlassian.net/browse/AAE-37399?atlOrigin=eyJpIjoiMmI4NmM1YWFkMjYxNDFmMThhZDFhNmVjNzgxNWRhZTQiLCJwIjoiaiJ9)

<!--
If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
-->

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

mcoune committed:
Adds root process instance ID to CloudTask

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Activiti/activiti-cloud?pullRequest=1904) <br/>All committers have signed the CLA.

mcoune committed:
Fixes typo in CloudTaskImpl

fkindgen committed:
Speed up build by skipping tests

fkindgen commented on .github/workflows/main.yml:
```diff
@@ -106,7 +106,7 @@ jobs:
       - name: Maven Build and Test
         shell: bash
-        run: mvn -DskipAcceptanceTests=true -DunitTests.parallel=true -T 1C install ${{ env.MAVEN_CLI_OPTS}} ${{ env.SONAR_SCANNER_OPTS }}
+        run: mvn -DskipAcceptanceTests=true -DskipTests -DskipITs -T 8C install ${{ env.MAVEN_CLI_OPTS}} ${{ env.SONAR_SCANNER_OPTS }}
```
Please revert before merge

fkindgen commented on .github/workflows/main.yml:
```diff
@@ -142,7 +142,7 @@ jobs:
   acceptance-tests:
     runs-on: ubuntu-latest
-    if: ${{ needs.pre-checks.outputs.skip != 'true' }}
+    if: false
```
please revert before merge

mcoune committed:
Adds root process instance id to CloudTask

mcoune committed:
Merge remote-tracking branch 'origin/feature/AAE-37399-add-root-process-instance-id-to-task' into feature/AAE-37399-add-root-process-instance-id-to-task

fkindgen committed:
Update main.yml

fkindgen commented on .github/workflows/main.yml:
```diff
@@ -339,7 +339,6 @@ jobs:
     needs:
       - pre-checks
       - build
-      - delete-test-images
```
Revert before merge

mcoune committed:
Renames root process instance id property

mcoune committed:
Merge remote-tracking branch 'origin/feature/AAE-37399-add-root-process-instance-id-to-task' into feature/AAE-37399-add-root-process-instance-id-to-task

mcoune committed:
prettier

mcoune committed:
Renames getter for root process instance id

mcoune committed:
unprettier

mcoune committed:
prettier

mkrbr committed:
Updates activiti.version to snapshot

Updates the activiti.version property to the 0.0.1-PR-5111-1902-SNAPSHOT version across multiple modules. This ensures that the projects use the correct snapshot version for development and testing.

mcoune committed:
Adds logging to CloudTaskEventImpl

mcoune committed:
Merge remote-tracking branch 'origin/feature/AAE-37399-add-root-process-instance-id-to-task' into feature/AAE-37399-add-root-process-instance-id-to-task

mcoune committed:
Renames task process root instance id property

Renames `taskProcessRootProcessInstanceId` to `rootProcessInstanceId`
for better consistency and clarity across the task model.

This change improves readability and maintainability by using a more
concise and widely understood name for the root process instance ID
associated with a task.

mcoune committed:
Sets root process instance id for tasks

mcoune committed:
Removes setting parent process instance id

mcoune committed:
Updates task entity root process instance id

mcoune committed:
Removes unnecessary slf4j dependency

mcoune committed:
Removes commented-out code

mcoune committed:
Sets parent process instance id in task event

mcoune committed:
reverted changes

mcoune committed:
reverted changes

mcoune commented on activiti-cloud-api/pom.xml:
```diff
@@ -12,7 +12,7 @@
   <packaging>pom</packaging>
   <properties>
-    <activiti.version>8.8.0-alpha.86</activiti.version>
+    <activiti.version>0.0.1-PR-5111-1902-SNAPSHOT</activiti.version>
```
Change to new stable version

mcoune committed:
reverted CI changes

mcoune committed:
Ensures root process instance id is audited

mcoune committed:
prettier

mcoune committed:
Adds task process root instance id to swagger

mcoune committed:
Sets root process instance id for task

mcoune committed:
removed unused import

mcoune committed:
moved getter and setter

mcoune committed:
Updates activiti.version property

mcoune committed:
Merge branch 'develop' into feature/AAE-37399-add-root-process-instance-id-to-task

mcoune committed:
Updates activiti.version property

mcoune committed:
Merge branch 'develop' into feature/AAE-37399-add-root-process-instance-id-to-task
